File: Programmers/pro_12909_올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
        if(i=='('):
            stack.append(i)
        elif(not stack):
            stack.append("No!!")
            break
        else:
            if(stack[-1]!='('):
                break
            stack.pop()      
    return False if stack else True

File: Programmers/pro_12909_올바른괄호.js
function solution(s) {
    let stack = []
    for (let i = 0; i < s.length; i++) {
        let len = stack.length
        let char = s[i]
        if (char == "(")
            stack.push(char)
        else if (len && stack[len - 1] == "(")
            stack.pop()
        else
            return false
    }
    return stack.length ? false : true
}